documentation/correctness.adoc

```diff
@@ -0,0 +1,123 @@
+. How is an incorrectness to be reported?
+. Can an incorrectness be recovered from (and if so: how)?
+
+
+== Serialization chunk
+
+=== Low-level syntactic
+
+The serialization is parsable as JSON.
+Note that “parsable” doesn't necessarily mean “syntax-correct”.
+E.g., trailing commas are not syntax-correct, but don't necessarily prevent the JSON from being parsable, nor deserializable.
+It's desirable to be able to recover from "`simple`"/"`obvious`" syntax errors in the JSON.
+
+Incorrectness is to be reported in terms of the location where parsing the JSON text failed, including an indication whether this was recoverable.
+The reporting is typically dependent on the precise JSON parser used.
+
+=== Structural syntactic
+
+The JSON conforms to link:../schemas/serialization.schema.json[the JSON Schema for the serialization format]footnote:[This file should match link:https://github.com/LIonWeb-org/lioncore-typescript/blob/main/schemas/generic-serialization.schema.json[the one in the `lioncore-typescript` repository].].
+The `serializationFormatVersion` property of the root object of the serialization indicates which version to use (or should do so).
+Currently, there's only one version, which is identified as: `"1"` (i.e., a string containing the integer 1).
+
+Note that JSON Schema isn't powerful enough to express more high-level, nor constraints of a referential nature on the serialization chunk.
+
+Incorrectness is to be reported in terms of e.g. a https://github.com/json-path/JsonPath[JsonPath] to the location(s) of failure(s).
+This is typically dependent on the precise JSON Schema validator framework/library/utility used.
+Recovering from an incorrectness is sometimes possible, e.g.:
+
+. If extra properties are present, they can simply be removed or ignored.
+. If properties are missing, they could just be added, provided they have a compliant default value
+
+=== Hierarchical
+
+The JSON doesn't violate the following constraints.
+
+. IDs of serialized nodes are unique (within the serialization).
+Violation is to be reported in terms of precise locations in the JSON file of all the serialized nodes with equal ID.
+. The `parent` property of a serialized node matches the `id` of the containing node (whenever that containing node is present in the serialization chunk).
+Concretely: if a node `N` is a child of a node `P` then `N.parent` should equal `P`.
+Violation is to be reported in terms of the ID of the contain**ing** serialized node, the ID of the contain**ed** serialized node, and the `parent` ID stated by that.
+. No duplicate containments, i.e. nodes are not the children of multiple other nodes, nor of themselves.
+Violation is to be reported in terms of the ID of the duplicately-contained serialized node, and the IDs of all serialized nodes that claim to contain it, including the links by which this happens, as well the index in the value of those links.
+. **TODO**  extend for annotations
```

Comment:
I tend to use `# TODO: ... #` formatting for TODOs, they show up with yellow background.